app/controllers/coronavirus_form/check_answers_controller.rb

```diff
@@ -9,7 +9,16 @@ def show
   end
 
   def submit
-    FormResponse.create(form_response: session) unless smoke_tester?
+    unless smoke_tester?
+      form_response = FormResponse.new(form_response: session.to_hash)
+
+      begin
+        form_response.save!
+      rescue ActiveRecord::RecordInvalid => e
+        GovukError.notify(e.message)
+        form_response.save(validate: false)
+      end
+    end
```

Comment:
This will work 👍Though, would this currently permit invalid data for any field?

Since this is to ensure we don't prevent users from submitting valid data, perhaps we could check that the form_response is valid against the JSON schema outside of an active record validation.

This could happen in the controller, or perhaps in a validation helper that always passes but sends an error to Sentry if the data doesn't pass the schema.

For example, search-api tests that [a content item is valid against a schema](https://github.com/alphagov/search-api/blob/4dd65e29239d8b3f878ffc93cabd76b10e0de8bd/spec/unit/content_item_publisher/finder_presenter_spec.rb#L25). This is done in a test, but I reckon we could do the same assertion in a controller, e.g.

```ruby
# in check your answers controller, before FormResponse.new(...).save!
unless valid_against_schema(form_response)
  GovukError.notify("invalid")
end
```

Later, once we're happy we're not going to stop people from submitting valid responses, we could introduce an active record validation for this data. What do you think?

Reply:
Sure! I've made the changes and fixed up. As the validation has been separated from the model, I don't have to change how the session is stored, that means there's no risk of any changes to the data export.